src/core/Akka/IO/Tcp.cs

```diff
@@ -806,21 +622,25 @@ public override string ToString() =>
         /// </summary>
         public sealed class CommandFailed : Event
         {
-            /// <summary>
-            /// TBD
-            /// </summary>
-            /// <param name="cmd">TBD</param>
-            public CommandFailed(Command cmd) => Cmd = cmd;
+            public CommandFailed(Command cmd, Option<Exception> ex)
+            {
+                Cmd = cmd;
+                Cause = ex;
+            }
+
+            public CommandFailed(Command cmd) : this(cmd, Option<Exception>.None)
+            {
+            }
 
             /// <summary>
-            /// TBD
+            /// The original command which failed.
             /// </summary>
             public Command Cmd { get; }
 
             /// <summary>
             /// Optionally contains the cause why the command failed.
             /// </summary>
-            public Option<Exception> Cause { get; private set; } = Option<Exception>.None;
+            public Option<Exception> Cause { get; private set; }
```

Comment:
shouldn't this have no setter and the property should be set by the constructor?
Also, shouldn't it be set to `None` by default?

Reply:
The compiler told me that it was unnecessary to set this so I opted not to, but maybe that's a mistake.

Reply:
Ah, I know why this isn't necessary - we always set via the CTOR now. Also, I've removed the private setter on this property too.